src/main/c/Makefile

```diff
@@ -24,6 +24,7 @@ libbwa.$(LIB_EXT): $(JNI_BASE_NAME).o jnibwa.o bwa/libbwa.a
 
 bwa:
 	git clone https://github.com/lh3/bwa && cd bwa && git checkout $(BWA_MEM_COMMIT) && echo '#define BWA_COMMIT "'$(BWA_MEM_COMMIT)'"' > bwa_commit.h
+	sed -i .bak 's/\(LOBJS=.*\)/\1 $$(AOBJS)/g' bwa/Makefile
```

Comment:
I had just copied ```bwa_index``` into jnibwa.c change its name to ```custom_bwa_index```, add all the includes in ```bwtindex.c``` and change the call to ```bwa_index``` to ```custom_bwa_index```. So the idea is to start just replicating bwa_index and slowly removing the bits that we don't want such as opt-parsing... However I get the SEG-FAULT even without any changes to the code!

I have tried to change our Makefile flag variables to match those of bwa's but no improvement.

What is going on!? Any ideas welcome.

Next thing I could try is to compile ```custom_bwa_index``` within bwa so that is available in libbwa.a, perhaps that "fixes" it but that is not the route we would like to take.